test/commands.test.ts

```diff
@@ -0,0 +1,259 @@
+import * as vscode from 'vscode';
+import * as assert from 'assert';
+
+interface TestEditorOptions {
+    language?: string;
+    content?: string;
+}
+
+type TestEditorAction = (editor: vscode.TextEditor, document: vscode.TextDocument) => void;
+
+async function inTextEditor(options: TestEditorOptions, action: TestEditorAction) {
+    const d = await vscode.workspace.openTextDocument(options);
+    await vscode.window.showTextDocument(d);
+    await action(vscode.window.activeTextEditor!, d);
+}
+
+function move(editor: vscode.TextEditor, line: number, col: number) {
+    const pos = new vscode.Position(line, col);
+    editor.selection = new vscode.Selection(pos, pos);
+}
+
+function select(editor: vscode.TextEditor, range: vscode.Range) {
+    editor.selection = new vscode.Selection(range.start, range.end);
+}
+
+const weekdayArray = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+
+suite('Commands', () => {
+
+    test('Demote', async () => {
+        const steps = [
+            '* Home',
+            '** Home',
+            '*** Home',
+            '**** Home',
+            '***** Home',
+        ];
+
+        await inTextEditor({ language: 'org', content: steps[0] }, async (e, d) => {
+            for (let i = 1; i < steps.length; ++i) {
+                await vscode.commands.executeCommand('org.doDemote');
+                assert.equal(d.getText(), steps[i]);
+            }
+        });
+    }).timeout(10000); // First test could be slow because of VSCode init
```

Comment:
Is there a default timeout that you overwrite?

Reply:
We can decrease the time of the first test by adding the following argument in .vscode/launch.json
 […]
{
            "name": "Launch Tests",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": [ **"--disable-extensions",**"--extensionDevelopmentPath=${workspaceRoot}", "--extensionTestsPath=${workspaceRoot}/out/test" ],
            "stopOnEntry": false,
            […]

see : https://code.visualstudio.com/docs/extensions/testing-extensions#_disabling-other-extensions